src/common/hooks/useConnection.ts

```diff
@@ -1,68 +1,64 @@
-import { useContext, useEffect } from "react";
-import {
-  ConnectionContext,
-  EMPTY,
-  actions,
-} from "common/context/ConnectionContext";
+import { useContext, useEffect, useState } from "react";
+import { ConnectionContext, actions } from "common/context/ConnectionContext";
 import { ethers } from "ethers";
 import { Wallet } from "bnc-onboard/dist/src/interfaces";
 import { SUPPORTED_NETWORK_IDS } from "common/config";
 
-export function useConnection() {
+export default function useConnection() {
+  const [initOnboard, setInitOnboard] = useState(false);
   const context = useContext(ConnectionContext);
-  if (context === EMPTY) {
+  if (!Object.keys(context).length) {
```

Comment:
Yup, its initialized as an empty object as it's initial state instead of this EMPTY symbol thing.

Reply:
I switched because it was causing some nasty type errors elsewhere in the app, as technically this type could be Symbol OR the Context object we expect... and even checking if it was EMPTY didn't allow me to check if it had properties.

Reply:
Would typing the context as `TContext | undefined` fix this? We could do the check on undefined here then

Reply:
I'm not 100% sure. Because even when I did a "null" check, it still didn't allow me to check for the state properties context should have.

I think it may have something to do with the Symbol type you create here.